serializer/src/schema.c

```diff
@@ -1808,11 +1831,18 @@ SCHEMA_STRUCT_TYPE_HANDLE Schema_CreateStructType(SCHEMA_HANDLE schemaHandle, co
                 /* If possible, reduce the memory of over allocation */
                 if (result == NULL)
                 {
-                    SCHEMA_STRUCT_TYPE_HANDLE* oldStructTypes = (SCHEMA_STRUCT_TYPE_HANDLE*)realloc(schema->StructTypes, sizeof(SCHEMA_STRUCT_TYPE_HANDLE) * schema->StructTypeCount);
-                    if (oldStructTypes == NULL)
+                    if (schema->StructTypeCount > 0)
                     {
-                        result = NULL;
-                        LogError("(Error code:%s)", ENUM_TO_STRING(SCHEMA_RESULT, SCHEMA_ERROR));
+                        SCHEMA_STRUCT_TYPE_HANDLE *oldStructTypes = (SCHEMA_STRUCT_TYPE_HANDLE *)realloc(schema->StructTypes, sizeof(SCHEMA_STRUCT_TYPE_HANDLE) * schema->StructTypeCount);
+                        if (oldStructTypes == NULL)
+                        {
+                            result = NULL;
+                            LogError("(Error code:%s)", ENUM_TO_STRING(SCHEMA_RESULT, SCHEMA_ERROR));
+                        }
+                        else
+                        {
+                            schema->StructTypes = oldStructTypes;
+                        }
```

Comment:
This solution wouldn't work. Did you compile the code?
Variable oldStructTypes doesn't exist inside the block below.

Reply:
I would suggest using this construction to all these blocks (taking the "properties" block as example):

```c
if (result != SCHEMA_OK)
{
    if (modelType->PropertyCount == 0)
    {
        modelType->Properties = NULL;
    }
    else
    {
        SCHEMA_PROPERTY_HANDLE *oldProperties = (SCHEMA_PROPERTY_HANDLE *)realloc(modelType->Properties, sizeof(SCHEMA_PROPERTY_HANDLE) * modelType->PropertyCount);
        if (oldProperties == NULL)
        {
            result = SCHEMA_ERROR;
            LogError("(result = %s)", ENUM_TO_STRING(SCHEMA_RESULT, result));
        }
        else
        {
            modelType->Properties = oldProperties;
        }
    
    }
}
```

Because, as you can see [here](https://github.com/Azure/azure-iot-sdk-c/blob/e27f1a22c7a9db1441a171b3250082952a198fe9/serializer/src/schema.c#L710), initially when propertyCount is 0 properties is NULL.

Reply:
The other option is, still related to properties as an example, whenever ```free(newProperty);``` appears just add ```modelType->Properties = NULL;``` afterwards. 
That actually seems to be a better solution.